src/core/gltf2importer/gltf2uri.cpp

```diff
@@ -0,0 +1,93 @@
+/*
+    gltf2uri.cpp
+
+    This file is part of Kuesa.
+
+    Copyright (C) 2019 Klarälvdalens Datakonsult AB, a KDAB Group company, info@kdab.com
+    Author: Jean-Michaël Celerier <jean-michael.celerier@kdab.com>
+
+    Licensees holding valid proprietary KDAB Kuesa licenses may use this file in
+    accordance with the Kuesa Enterprise License Agreement provided with the Software in the
+    LICENSE.KUESA.ENTERPRISE file.
+
+    Contact info@kdab.com if any conditions of this licensing are not clear to you.
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as
+    published by the Free Software Foundation, either version 3 of the
+    License, or (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "gltf2uri_p.h"
+#include "gltf2context_p.h"
+#include "kuesa_p.h"
+
+QT_BEGIN_NAMESPACE
+
+namespace Kuesa {
+namespace GLTF2Import {
+UriKind uriKind(const QString &uri)
+{
+    return uri.startsWith(QLatin1Literal("data:"), Qt::CaseInsensitive)
+            ? UriKind::Data
+            : UriKind::Path;
+}
+
+QUrl absoluteUrlFromGLTFUri(const QString &uriString, const QDir &basePath)
+{
+    QUrl url;
+    if (uriString.startsWith(QLatin1String("qrc:"), Qt::CaseInsensitive)) {
+        url = QUrl(uriString);
+    } else if (uriString.startsWith(QLatin1String("file:"), Qt::CaseInsensitive)) {
+        url = QUrl::fromLocalFile(basePath.absoluteFilePath(QUrl(uriString).toLocalFile()));
+    } else {
+        const QString absolutePath = basePath.absoluteFilePath(uriString);
+
+        // Handling the case of Qt resources
+        // QUrl(":/path") actually gives QUrl("")
+        // However QUrl("qrc:/path) is valid
+        if (absolutePath.startsWith(QLatin1String(":/")))
+            url = QUrl(QStringLiteral("qrc") + absolutePath);
+        else
+            url = QUrl::fromLocalFile(absolutePath);
+    }
+    return url;
+}
+
+QString localFileFromGLTFUri(const QString &uri, const QDir &basePath)
+{
+    return absoluteUrlFromGLTFUri(uri, basePath).toLocalFile();
+}
+
+QByteArray dataFromUri(const QString &uri, const QDir &basePath, GLTF2Context *context, bool &success)
```

Comment:
Why do we need the context to parse an uri?

Reply:
place to share code... this is used in multiple places

Reply:
hmm, I just checked and the GLTF2Context::parseUri method can be put static so certainly the need for a context instance could be removed

Reply:
maybe all of this should go in a small Uri namespace ?